extensions/amp-a4a/0.1/real-time-config-manager.js

```diff
@@ -65,331 +70,376 @@ export const RTC_ERROR_ENUM = {
+   * per-network macros.
+   * @param {!Object<string, !../../../src/service/variable-source.AsyncResolverDef>} macros
+   */
+  assignMacros(macros) {
+    macros['TIMEOUT'] = () => this.rtcConfig_['timeoutMillis'];
+    return macros;
+  }
+
+  /**
+   * Manages sending the RTC callouts for the Custom URLs.
+   * @param {!Object<string, !../../../src/service/variable-source.AsyncResolverDef>} customMacros The ad-network specified macro
+   */
+  handleRtcForCustomUrls(customMacros) {
+    // For each publisher defined URL, inflate the url using the macros,
+    // and send the RTC request.
+    (this.rtcConfig_.urls || []).forEach(urlObj => {
+      let url, errorReportingUrl;
+      if (isObject(urlObj)) {
+        url = urlObj.url;
+        errorReportingUrl = urlObj.errorReportingUrl;
+      } else if (typeof urlObj == 'string') {
+        url = urlObj;
       } else {
-        user().warn(TAG, `Unknown macro: ${macro} for vendor: ${vendor}`);
+        dev().warn(TAG, `Invalid url: ${urlObj}`);
       }
+      this.inflateAndSendRtc_(url,
+          customMacros,
+          errorReportingUrl);
     });
-    // The ad network defined macros override vendor defined/pub specifed.
-    const macros = Object.assign(validVendorMacros, customMacros);
-    inflateAndSendRtc_(a4aElement, url, seenUrls, promiseArray, rtcStartTime,
-        macros, rtcConfig['timeoutMillis'], errorReportingUrl,
-        vendor.toLowerCase());
-  });
-  return Promise.all(promiseArray);
-}
+  }
 
-/**
- * @param {!./amp-a4a.AmpA4A} a4aElement
- * @param {string} url
- * @param {!Object<string, boolean>} seenUrls
- * @param {!Array<!Promise<!rtcResponseDef>>} promiseArray
- * @param {number} rtcStartTime
- * @param {!Object<string, !../../../src/service/variable-source.AsyncResolverDef>} macros
- * @param {number} timeoutMillis
- * @param {string} errorReportingUrl
- * @param {string=} opt_vendor
- * @private
- */
-export function inflateAndSendRtc_(a4aElement, url, seenUrls, promiseArray,
-  rtcStartTime, macros, timeoutMillis, errorReportingUrl, opt_vendor) {
-  const {win} = a4aElement;
-  const ampDoc = a4aElement.getAmpDoc();
-  const callout = opt_vendor || getCalloutParam_(url);
-  const checkStillCurrent = a4aElement.verifyStillCurrent.bind(a4aElement)();
+  /**
+   * Manages sending the RTC callouts for all specified vendors.
+   * @param {!Object<string, !../../../src/service/variable-source.AsyncResolverDef>} customMacros The ad-network specified macro
+   */
+  handleRtcForVendorUrls(customMacros) {
+    // For each vendor the publisher has specified, inflate the vendor
+    // url if it exists, and send the RTC request.
+    Object.keys(this.rtcConfig_.vendors || []).forEach(vendor => {
+      const vendorObject = RTC_VENDORS[vendor.toLowerCase()];
+      const url = vendorObject ? vendorObject.url : '';
+      const errorReportingUrl = vendorObject ?
+        vendorObject['errorReportingUrl'] : '';
```

Comment:
you're switching between dot and array access of fields, change to vendorObject.errorReportingUrl  Also you can move this line after the if (!url) check